### ISSUE TITLE
Un(der?)documented `--tests` option and requirement for it

### ISSUE THREAD

**Issue by AMDmi3** [rust-lang/rust-clippy#16111]
### Description

I've been struggling to make `ignore_without_reason` lint to work for my code, and it turned out that it won't work even for corresponding ui test from clippy repository:

```rust
#![warn(clippy::ignore_without_reason)]

fn main() {}

#[test]
fn unignored_test() {}

#[test]
#[ignore = "Some good reason"]
fn ignored_with_reason() {}

#[test]
#[ignore] //~ ignore_without_reason
fn ignored_without_reason() {}
```

It took some time to discover that no lints work for test code at all, and then to discover undocumented `--tests` option by pure chance.

Is it expected that the option and the fact that it's required to lint tests is not documented or mentioned anywhere? IMO it is not right that https://rust-lang.github.io/rust-clippy/master/index.html#ignore_without_reason mention example which does not work.

### Version

```text

```

**Comment by Alexendoo** [comment]
It's inherited from `cargo check`, `cargo clippy --help` mentions `See all options with cargo check --help` but it's in the middle of the message so not the most discoverable, we probably should just repeat all the applicable flags in our own `--help` message

Mentioning it in lints that only apply to tests also is a good idea

**Comment by AMDmi3** [comment]
> It's inherited from cargo check, cargo clippy --help

I see it now, thanks!

Well, not sure if duplicating all inherited arguments is really a best solution, but my case should be fixed by mentioning an extra required argument in the lint(s) documentation. See #16205
